test/functional/admin/edition_images_controller_test.rb

```diff
@@ -0,0 +1,4 @@
+require "test_helper"
+
+class Admin::EditionImagesControllerTest < ActionDispatch::IntegrationTest
+end
```

Comment:
It would be nice to cover the `redirect_unless_user_can_preview_images_update` and `enforce_permissions!` behaviour here.

There is some prior art for testing permissions in other controllers, for example the attachments controller test:

https://github.com/alphagov/whitehall/blob/4e087da57a78a4fb29cb0ae5b57db91e142e38e8/test/functional/admin/attachments_controller_test.rb#L442-L446

But I don't think this test will copy-and-paste nicely. 😕  The attachments controller test is a 'functional style' test that inherits from `ActionController::TestCase`. This type of test [is discouraged nowadays](https://api.rubyonrails.org/classes/ActionController/TestCase.html).

https://github.com/alphagov/whitehall/blob/4e087da57a78a4fb29cb0ae5b57db91e142e38e8/test/functional/admin/attachments_controller_test.rb#L3

Whereas the images controller test inherits from `ActionDispatch::IntegrationTest` because it's an 'integration style' test. This is the recommended approach and default Rails behaviour when creating new controller tests, which is how you've ended up with it here.

https://github.com/alphagov/whitehall/blob/f506f75b0d95d132558980da7dfa9c6f08239971/test/functional/admin/edition_images_controller_test.rb#L3

These different types of tests operate in slightly different contexts, so the way you write tests differs too.

I think something like this should work:

```ruby
edition = create(:draft_case_study)
login_as create(:world_editor)
get admin_edition_images_path(edition.id)
assert_equal 403, status
```